test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs

```diff
@@ -411,6 +411,191 @@ public void SourcesCommandTest_AddWithUserNamePassword_UserDefinedConfigFile()
             }
         }
 
+        [Theory]
+        [InlineData(@"https://source.test", true)]
+        [InlineData(@"\\myserver\packages", false)]
+        public void SourcesCommandTest_AddWithProtocolVersion(string source, bool shouldWriteProtocolVersion)
```

Comment:
I've pushed a change that should fix tests failing on Mac. Switched to using `Util.VerifyResultSuccess` which is used in other tests instead of `VerifyResultFailure`.

As for behaviour with local feeds, I've implemented it as per [this comment](https://github.com/NuGet/NuGet.Client/pull/5002#pullrequestreview-1287837289).

Reply:
I believe that the reason why test failed on Mac was due to `\\myserver\packages` not being treated as local source. I've split the test into two: one for remote, one for local. Test for local source uses different source path depending on the platform it runs on.

Reply:
My bad! I updated only tests for `update` and forgot about `add`. I've just pushed updated tests for `add`.